### PR TITLE
Support adding vhost-net network devices

### PR DESCRIPTION
**Pull request by justxuewei**
Argument `virnets` receives a JSON string of `NetworkInterfaceConfig`, rather than `VirtioNetDeviceConfigInfo`. It allows user to add any network devices according to item's backend type.

### CODE REVIEW

File: src/api_server.rs [hunk 1]
```diff
@@ -85,11 +84,11 @@ impl ApiServer {
...
                     None => return Err(anyhow!("The config of virtio-net device is required")),
                 };
-                let configs: Vec<VirtioNetDeviceConfigInfo> = serde_json::from_str(config_json)
-                    .context("Parse virtio-net device config from json")?;
+                let configs: Vec<NetworkInterfaceConfig> = serde_json::from_str(config_json)
+                    .context("Parse NetworkInterfaceConfig from JSON")?;
                 for config in configs.iter() {
                     self.insert_virnet(config.clone())
-                        .context("Insert a virtio-net device to the Dragonball")?;
+                        .context("Insert a virtio device to the Dragonball")?;
```
Comment (studychao) [inline src/api_server.rs]: The user cannot know that it's net device that went wrong in this error message.

File: src/vmm_comm_trait.rs [hunk 2]
```diff
@@ -135,9 +134,9 @@ pub trait VMMComm {
         Ok(())
     }
 
-    fn insert_virnet(&self, config: VirtioNetDeviceConfigInfo) -> Result<()> {
+    fn insert_virnet(&self, config: NetworkInterfaceConfig) -> Result<()> {
         self.handle_request(Request::Sync(VmmAction::InsertNetworkDevice(config)))
-            .context("Request to insert a virtio-net device")?;
+            .context("Request to insert a virtio device")?;
```
Comment (studychao) [inline src/vmm_comm_trait.rs]: ditto

File: src/api_server.rs (hunk 1 above)
Comment (justxuewei) [reply src/api_server.rs]: Fixed.

File: src/vmm_comm_trait.rs (hunk 2 above)
Comment (justxuewei) [reply src/vmm_comm_trait.rs]: Fixed.

**Review by studychao — APPROVED** [review]

**Pull request merged**